crates/weaver_health/src/attribute_advice.rs

```diff
@@ -0,0 +1,308 @@
+// Pluggable advisors
+
+use convert_case::{Boundary, Case, Casing};
+use serde::Serialize;
+use serde_json::Value;
+use weaver_resolved_schema::attribute::Attribute;
+use weaver_semconv::{
+    attribute::{AttributeType, PrimitiveOrArrayTypeSpec, TemplateTypeSpec, ValueSpec},
+    stability::Stability,
+};
+
+use crate::{attribute_health::AttributeHealthChecker, sample::SampleAttribute};
+
+/// The advisory level of an advice
+#[derive(Debug, Clone, PartialEq, Serialize, PartialOrd, Ord, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum Advisory {
+    /// Useful context without action needed
+    Information,
+    /// Suggested change that would improve things
+    Improvement,
+    /// Something that breaks compliance rules
+    Violation,
+}
+
+/// Represents a health check advice
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct Advice {
+    /// The key of the advice e.g. "is_deprecated"
+    pub key: String,
+    /// The value of the advice e.g. "true"
+    pub value: Value,
```

Comment:
I'm not sure I understand how this `value` field:
- is used by the reporting system,
- adds something compared to the following `message` field.

Reply:
My idea here is that:
- Key = the name of the advisor. So `stability` or `enum` or `type` etc. (perhaps `advisor_name` would be better than `key`)?
- Value = a machine readable simple indicator for the advice.
- Message = a human readable phrase more descriptive than the Value but to variable too be machine readable.